src/Transport/Sending/OutgoingMessageExtensions.cs

```diff
@@ -51,17 +47,31 @@ static void ApplyDeliveryConstraints(ServiceBusMessage message, DispatchProperti
             }
         }
 
-        static void ApplyCorrelationId(ServiceBusMessage message, Dictionary<string, string> headers)
+        static void ApplyCorrelationId(ServiceBusMessage message, IOutgoingTransportOperation outgoingTransportOperation)
         {
-            if (headers.TryGetValue(Headers.CorrelationId, out var correlationId))
+            var properties = outgoingTransportOperation.Properties;
+            var headers = outgoingTransportOperation.Message.Headers;
+
+            if (properties.TryGetValue(Headers.CorrelationId, out var correlationId))
```

Comment:
As far as I recall, the properties get serialized into the outbox. It might be worthwhile considering removing those temporary properties after reading them or add a comment why they need to be preserved

Reply:
Quick question. Technically, already today, it is possible to add a customization delegate that allows you to customize the entire native message created. [Any reason this can't be used](https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/blob/3dd31681a9e4fc42c5bcd549ed9f56b944cbf9e1/src/Transport/Sending/OutgoingTransportOperationExtensions.cs#L26)?

Reply:
And yes, it seems the customizer should also be removed from the properties dictionary because I doubt we can every serialize action delegates into outbox records.

Reply:
1. I didn't know that `TransportTransaction` was itself a `ContextBag` 🤯
2. Looked at the type you linked but I haven't figured out how that could be used as so far all invoked types I see are internal

So, I need additional research to validate if this could be used for the ******* connector.